sklearn/ensemble/base.py

```diff
@@ -178,3 +182,93 @@ def _partition_estimators(n_estimators, n_jobs):
     starts = np.cumsum(n_estimators_per_job)
 
     return n_jobs, n_estimators_per_job.tolist(), [0] + starts.tolist()
+
+
+class _BaseEnsembleHeterogeneousEstimator(MetaEstimatorMixin,
+                                          _BaseComposition,
+                                          BaseEstimator,
+                                          metaclass=ABCMeta):
```

Comment:
Does this need to be an abstract base class if it has no abstract methods?

Reply:
Oh in fact, `__init__` should be an `abstractmethod`. Good remark.